File: boletim-ocorrencia/src/main/java/br/edu/utfpr/td/tsi/sistema/boletim/ocorrencia/business/ReportRules.java
package br.edu.utfpr.td.tsi.sistema.boletim.ocorrencia.business;

import br.edu.utfpr.td.tsi.sistema.boletim.ocorrencia.model.StolenVehicleReport;

import java.util.List;

public interface ReportRules {
    public void register(StolenVehicleReport report);

    public List<StolenVehicleReport> readAll();
    public StolenVehicleReport readById(int id);
    public List<StolenVehicleReport> readByCity(String city);
    public List<StolenVehicleReport> readByDayTime(String dayTime);
    public void remove(int id);
}

File: boletim_python_v/occurance_report_system/occurance_report_system/occurance_report/busisness_rules/ReportRules.py
from abc import ABC, abstractmethod
from typing import List
from ..models import stolen_vehicle_report

class ReportRules(ABC):
    @abstractmethod
    def register(self, report: stolen_vehicle_report) -> None:
        pass

    @abstractmethod
    def read_all(self) -> List[stolen_vehicle_report]:
        pass

    @abstractmethod
    def read_by_id(self, id: int) -> stolen_vehicle_report:
        pass

    @abstractmethod
    def read_by_city(self, city: str) -> List[stolen_vehicle_report]:
        pass

    @abstractmethod
    def read_by_day_time(self, day_time: str) -> List[stolen_vehicle_report]:
        pass

    @abstractmethod
    def remove(self, id: int) -> None:
        pass
